overlord/snapstate/snapmgr_test.go

```diff
@@ -3173,6 +3173,99 @@ func (s *snapmgrTestSuite) TestRemoveRefusedLastRevision(c *C) {
 	c.Check(err, ErrorMatches, `snap "gadget" is not removable`)
 }
 
+func (s *snapmgrTestSuite) TestRemoveDeletesConfigOnLastRevision(c *C) {
+	si := snap.SideInfo{
+		RealName: "some-snap",
+		Revision: snap.R(7),
+	}
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	snapstate.Set(s.state, "some-snap", &snapstate.SnapState{
```

Comment:
Can we get a second snap here with some configuration to make sure we're not destroying the whole configuration set? That function might easily do that by mistake.